### PR TITLE
Updated dependencies to newest and Tesseract to 4.1.3

### PR DESCRIPTION
**Pull request by oskargargas**
To compile Tesseract without libcurl support autogen options were backported from 5.0.1 tag
Additional task to build only for iOS were added.

### Checklist
- I've validated my changes against the following plaftorms:
  - [x] iOS - Deployment Target Version 11.0
  - [] macOS Catalyst - Deployment Target Version 10.15
  - [] macOS - Deployment Target Version 10.13
- [] I've updated the documentation if necessary.
- [] I've read the [README](https://github.com/SwiftyTesseract/libtesseract/blob/main/README.md)
- [] I've read the [Contribution Guidelines](https://github.com/SwiftyTesseract/libtesseract/blob/main/CONTRIBUTING.md)
- [] I've read the [Code of Conduct](https://github.com/SwiftyTesseract/libtesseract/blob/main/CODE_OF_CONDUCT.md)

### Motivation and Context

Dependencies were outdated. They contained security vulnerabilities among other issues.
In the spirit of using the latest stable I also updated Tesseract to 4.1.3.
In future if need arises for mine project I will follow with PR updating Tesseract to 5.x

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -21,7 +22,7 @@ vars:
   TIFF_ARCHIVE: libtiff.a
   TIFF_CONFIG: "{{.TIFF_SRC}}/configure"
 
-  TESSERACT_VERSION: 4.1.3
+  TESSERACT_VERSION: 5.0.1
```
Comment (Steven0351) [inline Taskfile.yml]: Not really comfortable merging this until I can check if there are any API breakages for the use in SwiftyTesseract

**Comment by oskargargas** [comment]
Hm... to be honest I didn't expect that pushing to my branch will auto update this PR. 5.0.1 is not ready for merge for sure. I'll revert it back to 4.1.3.

**Comment by oskargargas** [comment]
Done, it's 4.1.3 again.

**Comment by Steven0351** [comment]
LGTM, I’m going to make sure the SwiftyTesseract tests pass before tagging a new version. Thanks for the contribution!

**Pull request merged**